### ISSUE TITLE
KingfisherManager.swift: Use of local variable 'handleNoCache' before its declaration

### ISSUE THREAD

**Issue by arismamo** [onevcat/Kingfisher#700]
Hi every body,
at first i want to thank the creator for this library.

**My problem is** 

I have used kingfisher in one of my iOS projects which i'm trying to update the swift version from 2.x -> 3.x. All other libraries are complied except Kingfisher which throws the exception "Use of local variable 'handleNoCache' before its declaration" in file KingfisherManager.swift

**My Pod file is:**

```
platform :ios, '8.0'
use_frameworks!
inhibit_all_warnings!
target 'app_name' do
	pod 'SCLAlertView'
	pod 'ActionSheetPicker-3.0'
	pod 'SVProgressHUD', '~> 2.1.2'
        pod 'RealmSwift'
	pod 'Alamofire', '~> 4.4'
	pod 'AlamofireObjectMapper', '~> 4.0'
	pod 'SugarRecord/CoreData'
    pod 'UIBarButtonItem-Badge', :git => 'https://github.com/mikeMTOL/UIBarButtonItem-Badge'
	pod 'Firebase', '~> 3.2.0'
	pod 'Firebase/Messaging'
    pod 'Kingfisher', '~> 3.0'
end
```

Can someone that had faced this problem, help me solve it?

Thank you in advance

**Comment by onevcat** [comment]
May I know what the Swift / Xcode version you are using?

**Comment by arkth-araya** [comment]
hey,
i'm facing the same problem too. i use carthage.

the error message is
'Kingfisher/Sources/KingfisherManager.swift:224:25: error: use of local variable 'handleNoCache' before its declaration
                        handleNoCache()'

so, you should move handleNoCache function to above line 212.

Maybe i will make pull-request later.

**Comment by arismamo** [comment]
@onevcat  I am using swift 3.0.2 and xcode 8.2.1

**Comment by arismamo** [comment]
@arkth-araya i can confirm that moving the function above the line, every works is fine. i think its due to the last version of swift. Thanks

**Comment by onevcat** [comment]
Seem it is a non-back-compatible change in Swift 3.1. I'll merge it and tag a new version now.

**Comment by onevcat** [comment]
Fixed in 3.10.1
